add option to not use sudo

Hades32 opened the pull request:
This should address Issue #66

alexellis commented:
The full template is needed including the evidence of testing. https://github.com/alexellis/k3sup/blob/master/.github/PULL_REQUEST_TEMPLATE.md

Take a moment to read the comment by Derek also. 

Thanks for the PR.

Hades32 commented:
## Description
Added the --sudo option which defaults to true (no functional change to previous behavior) which allows to disable the use of sudo in server commands.

## Motivation and Context
There was already issue #66. There are some servers that don't have sudo installed. If you have access to the root users, sudo is abviously not neccessary.


## How Has This Been Tested?
I tested with this environment successfully: https://gist.github.com/Hades32/1db58c0e9c55496a8c8524574b83da82

Creation of multiple clusters were successful.

## Types of changes
<!--- What types of changes does your code introduce? Put an `x` in all the boxes that apply: -->
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist:
<!--- Go over all the following points, and put an `x` in all the boxes that apply. -->
<!--- If you're unsure about any of these, don't hesitate to ask. We're here to help! -->
- [x] My code follows the code style of this project.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [x] I've read the [CONTRIBUTION](https://github.com/openfaas/faas/blob/master/CONTRIBUTING.md) guide
- [x] I have signed-off my commits with `git commit -s`
- [ ] I have added tests to cover my changes.
- [x] All new and existing tests passed.

I didn't update the readme as I assumed that not every seldom used command line parameter needs to go there.

Also no tests were added, as it didn't seem worth it, but happy to change if deemed necessary.
